mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot_backup_policy.go

```diff
@@ -0,0 +1,406 @@
+	if err := snapshotScheduleUpdate(d, conn, projectID, clusterName); err != nil {
+		return err
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"project_id":   projectID,
+		"cluster_name": clusterName,
+	}))
+
+	return resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyRead(d, meta)
+}
+
+func resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	//Get client connection.
+	conn := meta.(*matlas.Client)
+
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	backupPolicy, _, err := conn.CloudProviderSnapshotBackupPolicies.Get(context.Background(), projectID, clusterName)
+	if err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicyRead, clusterName, err)
+	}
+
+	if err := d.Set("cluster_id", backupPolicy.ClusterID); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "cluster_id", clusterName, err)
+	}
+	if err := d.Set("reference_hour_of_day", backupPolicy.ReferenceHourOfDay); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "reference_hour_of_day", clusterName, err)
+	}
+	if err := d.Set("reference_minute_of_hour", backupPolicy.ReferenceMinuteOfHour); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "reference_minute_of_hour", clusterName, err)
+	}
+	if err := d.Set("restore_window_days", backupPolicy.RestoreWindowDays); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "restore_window_days", clusterName, err)
+	}
+	if err := d.Set("update_snapshots", backupPolicy.UpdateSnapshots); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "update_snapshots", clusterName, err)
+	}
+	if err := d.Set("next_snapshot", backupPolicy.NextSnapshot); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "next_snapshot", clusterName, err)
+	}
+	if err := d.Set("policies", flattenPolicies(backupPolicy.Policies)); err != nil {
+		return fmt.Errorf(errorSnapshotBackupPolicySetting, "policies", clusterName, err)
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*matlas.Client)
+
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	if err := snapshotScheduleUpdate(d, conn, projectID, clusterName); err != nil {
+		return err
+	}
+
+	return resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyRead(d, meta)
+}
+
+func resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyDelete(d *schema.ResourceData, meta interface{}) error {
+	// there is no way to det the default configuration
+	return nil
+}
+
+func resourceMongoDBAtlasCloudProviderSnapshotBackupPolicyImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conn := meta.(*matlas.Client)
+
+	parts := strings.SplitN(d.Id(), "-", 2)
```

Comment:
Yes don't worry about it, in this resource, we won't have problems because the clusterName is in the last position, so the "strings.SplitN(d.Id(), "-", 2)" function parts it in two string when it catches the first dash coincidence ("-"), in this case, it is in before the clusterName, the problem is when there are three variables and the variable in the middle has dashes.
I thought it and then tested it while I developed the resource 👍

Reply:
Awesome, thank you for ensuring and the explanation.